docs/source/recipes.rst

```diff
@@ -253,21 +253,21 @@ You could run this operation directly with: ``mpiexec -n 2 python project.py run
 MPI-operations with ``flow.cmd``
 --------------------------------
 
-Alternatively, you can implement an MPI-parallelized operation with the ``flow.cmd`` decorator, optionally in combination with the ``flow.directives`` decorator.
+Alternatively, you can implement an MPI-parallelized operation with the ``flow.cmd`` decorator, optionally in combination with the ``FlowProject.operation.with_directives`` decorator.
 This strategy lets you define the number of ranks directly within the code and is also the only possible strategy when integrating external programs without a Python interface.
 
 Assuming that we have an MPI-parallelized program named ``my_program``, which expects an input file as its first argument and which we want to run on two ranks, we could implement the operation like this:
 
 .. code-block:: python
 
-    @FlowProject.operation
+    @FlowProject.operation.with_directives({"np": 2})
     @flow.cmd
-    @flow.directives(np=2)
     def hello_mpi(job):
         return "mpiexec -n 2 mpi_program {job.ws}/input_file.txt"
 
 The ``flow.cmd`` decorator instructs **signac-flow** to interpret the operation as a command rather than a Python function.
-The ``flow.directives`` decorator provides additional instructions on how to execute this operation and is not strictly necessary for the example above to work.
+The ``@FlowProject.operation.with_directives(...)`` decorator provides additional instructions on how to execute this operation.
+The decorator ``@FlowProject.operation`` does not assign any directives to the operation.
```

Comment:
Wherever possible, we should link to the corresponding API docs, e.g. with
```suggestion
The decorator :py:meth:`FlowProject.operation` does not assign any directives to the operation.
```

Reply:
This needs to be mostly reworked in another PR to fix the out of data MPI documentation, so let's create an issue and fix that there.

Reply:
I might not have been clear enough. This comment applies to a lot of places not related to the MPI docs. I'll mark them.